pylearn2/datasets/svhn.py

```diff
@@ -46,44 +61,48 @@ class SVHN(dense_design_matrix.DenseDesignMatrixPyTables):
 
     def __init__(self, which_set, path=None, center=False, scale=False,
                  start=None, stop=None, axes=('b', 0, 1, 'c'),
-                 preprocessor = None):
+                 preprocessor = None, enable_cache=True,
+                 write_permission=False):
 
         assert which_set in self.mapper.keys()
 
         self.__dict__.update(locals())
         del self.self
 
-        if path is None:
-            path = '${PYLEARN2_DATA_PATH}/SVHN/format2/'
+        if write_permission and (not enable_cache):
+            if path is None:
+                raise ValueError("You are not allowed to change the original" +
+                                 " file located at $PYLEARN2_DATA_PATH. " +
+                                 " Enable caching to remove this error")
+            else:
+                warnings.warn("You might change the data in the files " +
+                              "located at: {}".format(path))
+
+        if not write_permission:
             mode = 'r'
         else:
             mode = 'r+'
-            warnings.warn("Because path is not same as PYLEARN2_DATA_PATH "
-                          "be aware that data might have been "
-                          "modified or pre-processed.")
-
-        if mode == 'r' and (scale or
-                            center or
-                            (start is not None) or
-                            (stop is not None)):
-            raise ValueError("Only for speed there is a copy of hdf5 file in "
-                             "PYLEARN2_DATA_PATH but it meant to be only "
-                             "readable. If you wish to modify the data, you "
-                             "should pass a local copy to the path argument.")
+            warnings.warn("Mode is set to write more, any changes you apply " +
```

Comment:
It looks like centering and scaling does modify data on disk. Same with preprocessor? Warnings should specify that.